presto-main/src/main/java/com/facebook/presto/sql/planner/PlanOptimizers.java

```diff
@@ -334,7 +335,8 @@ public PlanOptimizers(
                         estimatedExchangesCostCalculator,
                         ImmutableSet.of(
                                 new ImplementBernoulliSampleAsFilter(),
-                                new ImplementOffset())),
+                                new ImplementOffset(),
+                                new ImplementLimitWithTies(metadata))),
```

Comment:
We need this to be controlled by a session param/feature config. In fact, I would like the whole feature to be optional.